nixos/modules/tasks/filesystems/overlayfs.nix

```diff
@@ -0,0 +1,151 @@
+{ config, lib, pkgs, utils, ... }:
+
+let
+  toplevelConfig = config;
+
+  # Returns a service that creates the required directories before the mount is
+  # created.
+  preMountService = _name: fs:
+    let
+      rwName = "rw-${utils.escapeSystemdPath fs.mountPoint}";
+
+      # The scripted initrd contains some magic to add the prefix to the
+      # paths just in time, so we don't add it here.
```

Comment:
I wonder if we should *remove* this magic and add it here instead.

Reply:
Yeah I'm not opposed to removing the magic but doing so in this PR would make it much more critical than it needs to be.